src/main/java/sg/ncl/MainController.java

```diff
@@ -456,36 +449,67 @@ public String validateDetails(
             }
 
             if (errorsFound) {
+                logger.warn("Signup new team error {}", signUpMergedForm.toString());
                 return "/signup2";
             } else {
 
-                // FIXME need to check if team exists?
                 teamFields.put("name", signUpMergedForm.getTeamName());
                 teamFields.put("description", signUpMergedForm.getTeamDescription());
                 teamFields.put("website", signUpMergedForm.getTeamWebsite());
                 teamFields.put("organisationType", signUpMergedForm.getTeamOrganizationType());
                 teamFields.put("visibility", signUpMergedForm.getIsPublic());
                 mainObject.put("isJoinTeam", false);
-                registerUserToDeter(mainObject);
+
+                try {
+                    registerUserToDeter(mainObject);
+                } catch (TeamNotFoundException e) {
+                    redirectAttributes.addFlashAttribute("message", e.getMessage());
+                    return "redirect:/signup2";
+                } catch (Exception e) {
+                    redirectAttributes.addFlashAttribute("message", ERR_SERVER_OVERLOAD);
+                    return "redirect:/signup2";
+                }
+
+                logger.info("Signup new team success");
                 return "redirect:/team_application_submitted";
             }
         	
     	} else if (joinNewTeamName != null && !joinNewTeamName.isEmpty()) {
-
+            logger.info("Signup join team name {}", joinNewTeamName);
             // get the team JSON from team name
-            // FIXME need to check if team exists?
-            String teamIdToJoin = getTeamIdByName(signUpMergedForm.getJoinTeamName());
+            String teamIdToJoin = "";
+
+            try {
+                teamIdToJoin = getTeamIdByName(signUpMergedForm.getJoinTeamName());
+            } catch (TeamNotFoundException e) {
+                redirectAttributes.addFlashAttribute("message", e.getMessage());
+                return  "redirect:/signup2";
+            } catch (AdapterConnectionException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Combine this catch with the one at line 484, which has the same body. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2147)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)